c7n/resources/dynamodb.py

```diff
@@ -240,6 +240,94 @@ def process(self, resources):
                                 f.exception()))
 
 
+@Table.action_registry.register('backup')
+class CreateBackup(BaseAction, StatusFilter):
+    """Creates a manual backup of a DynamoDB table
+
+    :example:
+
+    .. code-block:: yaml
+
+            policies:
+              - name: dynamodb-create-backup
+                resource: dynamodb-table
+                actions:
+                  - type: backup
+    """
+
+    valid_status = ('ACTIVE',)
+    schema = type_schema('backup')
+    permissions = ('dynamodb:CreateBackup',)
+
+    def process(self, tables):
+        tables = self.filter_table_state(
+            tables, self.valid_status)
+        if not len(tables):
+            return
+
+        c = local_session(self.manager.session_factory).client('dynamodb')
+
+        for table_set in chunks(tables, 20):
+            for t in table_set:
+                arn = c.create_backup(
+                    BackupName=snapshot_identifier(
+                        'Backup', t['TableName']),
+                    TableName=t['TableName'])['BackupDetails']['BackupArn']
+                t['c7n:BackupArn'] = arn
+
+
+@resources.register('dynamodb-backup')
+class Backup(query.QueryResourceManager):
+
+    class resource_type(object):
+        service = 'dynamodb'
+        type = 'table'
+        enum_spec = ('list_backups', 'BackupSummaries', None)
+        detail_spec = None
+        id = 'Table'
+        filter_name = None
+        name = 'TableName'
+        date = 'BackupCreationDateTime'
+        dimension = 'TableName'
+        config_type = 'AWS::DynamoDB::Table'
+
+
+@Backup.action_registry.register('delete')
+class DeleteBackup(BaseAction):
+    """Deletes backups of a DynamoDB table
+
+    :example:
+
+    .. code-block:: yaml
+
+            policies:
+              - name: dynamodb-delete-backup
+                resource: dynamodb-backup
+                filters:
+                  - type: age
+                    days: 28
+                    op: ge
+                actions:
+                  - type: delete
+    """
+
+    schema = type_schema('delete')
+    permissions = ('dynamodb:DeleteBackup',)
+
+    def process(self, tables):
+
+        c = local_session(self.manager.session_factory).client('dynamodb')
+
+        for table_set in chunks(tables, 20):
```

Comment:
if your just going through the list items one by one can, just iterate the list, the notion of chunking and dispatching to a thread pool is to achieve higher concurrency, the chunking primarily to avoid high cross thread synchronizations when dealing with a large resource count.

one thing to also do is to catch on ClientError and check for some common error codes, table not found, backup with name already exists.